scheduler/test/cook/test/testutil.clj

```diff
@@ -237,3 +238,20 @@
                              {:level :info
                               :out (ConsoleAppender. (PatternLayout. "%d{ISO8601} %-5p %c [%t] - %m%n"))}))
   (log/info "Running" (:var m)))
+
+(let [minimal-config {:unhandled-exceptions {}
+                      :log {}
+                      :database {:datomic-uri ""}
+                      :mesos {:master ""
+                              :leader-path ""}
+                      :authorization {:one-user ""}
+                      :scheduler {}
+                      :zookeeper {:local? true}
+                      :port nil
+                      :metrics {}
+                      :nrepl {}}]
+  (defn setup
+    "Given an optional config map, initializes the config state"
+    [& {:keys [config], :or nil}]
+    (mount/stop)
+    (mount/start-with-args (merge minimal-config config) #'cook.config/config)))
```

Comment:
Do we want mount to initialize the entire dependency graph for each test we run? 

Is this less severe in the repl which triggers reloads of the `defstate`s from the loaded namespaces?

Reply:
> Do we want mount to initialize the entire dependency graph for each test we run?

I don't think so. I think we want each test to be specific about what it initializes. Currently, the `setup` function only initializes `cook.config/config`, but I expect it to change to allow for initializing other states as needed.

> Is this less severe in the repl which triggers reloads of the defstates from the loaded namespaces?

I don't understand this question, can you re-state?

Reply:
I want to see how it feels to use `with-redefs` with `mount` initialized state as that can be a common usage pattern.

Reply:
> I think we want each test to be specific about what it initializes.

But initializing an individual state will trigger upstream dependencies to have to be initialized first? I think being able to use `with-redefs` can avoid that concern.

Reply:
I added a unit test, `test-redef-config`, showing that a `defstate` can be `with-redef`'d, let me know what you think